CMakeLists.txt

```diff
@@ -15,7 +15,7 @@
 # specific language governing permissions and limitations
 # under the License.
 
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.16)
```

Comment:
Do we use any instructions that required 3.16? I saw fetch content instruction required 3.11+.

Reply:
Yeah, use 3.16 just because it is the default version in apt repo of ubuntu 20.04. But I highly recommend a newer version of cmake. [A good post](https://www.scivision.dev/cmake-changelog) described its advantages.

Reply:
OK, thanks for your explaination.

Reply:
@PragmaTwice what features we use for 3.16?

Reply:
i think 20.04 also is newer, we still use ubuntu 18.04 in CI. For some enterprise user, they even still use CentOS 6.